pkg/vm/builtins/ecop.go

```diff
@@ -36,6 +37,12 @@ func (e *EcOp) CheckWrite(segment *mem.Segment, offset uint64, value *mem.Memory
 }
 
 func (e *EcOp) InferValue(segment *mem.Segment, offset uint64) error {
+	// check if the value is already in the cache
+	value, ok := e.cache[offset]
+	if ok {
+		mv := mem.MemoryValueFromFieldElement(&value)
+		return segment.Write(offset, &mv)
+	}
```

Comment:
Is it possible that `InferValue` is called on the same offset multiple times? Because in that case the value might already be written in the segment.